### PR TITLE
Muriloo/source appstore

### PR DESCRIPTION
**Pull request by Muriloo**
## What
Adding Appstore as a source, as stated in this issue #1489 

## How
*Describe the solution*
It's based on the singer tap already created listed in the issue above.

## Pre-merge Checklist
- [ Yes ] *Run integration tests* => 2 tests fail because there's a field called '_time_extracted' that changes everytime a request is done. [This](https://airbytehq.slack.com/archives/C019WEENQRM/p1610481406167100) slack conversation describes it (check the few last messages)
- [ No ] *Publish Docker images* => I didn't know how to do it.

## Recommended reading order
1. airbyte-integrations/connectors/source-appstore/

### CODE REVIEW

File: airbyte-integrations/connectors/source-appstore-singer/source_appstore_singer/spec.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        "type": "string",
+        "description": "<a href=\"https://developer.apple.com/documentation/appstoreconnectapi/generating_tokens_for_api_requests\">Issuer_id</a> is used to generate the credentials to connect to appstore's API."
+      },
+      "vendor": {
+        "type": "string",
+        "description": "This is the Apple ID of your account."
+      },
+      "start_date": {
+        "type": "string",
+        "description": "describe me",
```
Comment (cgardens) [inline airbyte-integrations/connectors/source-appstore-singer/source_appstore_singer/spec.json:28]:
```suggestion
        "description": "date from which to start pulling data",
```

File: airbyte-integrations/connectors/source-appstore-singer/source_appstore_singer/spec.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
+{
+  "documentationUrl": "https://docsurl.com",
```
Comment (cgardens) [inline airbyte-integrations/connectors/source-appstore-singer/source_appstore_singer/spec.json:2]:
```suggestion
  "documentationUrl": "https://docs.airbyte.io/integrations/sources/appstore",
```

File: docs/integrations/sources/appstore.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
+# Braintree
```
Comment (cgardens) [inline docs/integrations/sources/appstore.md:1]:
```suggestion
# Appstore
```

File: airbyte-integrations/connectors/source-appstore-singer/source_appstore_singer/source.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        the properties of the spec.json file
+
+        :return: AirbyteConnectionStatus indicating a Success or Failure
+        """
+        try:
+            # create request fields for testing
+            api_fields_to_test = {
+                "subscription_report": {"reportType": "SUBSCRIPTION", "frequency": "DAILY", "reportSubType": "SUMMARY", "version": "1_2"}
+            }
+            test_date = date.today() - timedelta(days=2)
```
Comment (cgardens) [inline airbyte-integrations/connectors/source-appstore-singer/source_appstore_singer/source.py:56]: What will happen here if the person uses this has a new appstore account (not 2 days old)?

I'm wondering if we can make this a little easier and just check the http status code of the response. I looked at the [appstoreconnect docs](https://developer.apple.com/documentation/appstoreconnectapi/interpreting_and_handling_errors) and it looks like they use standard http codes. If we request a report for a date that hasn't been generated yet does it the API return a 200 and an empty report or does it return a 400? If it returns a 200, then we can just check that the http status code is good and not worry about trying to check that the report exists itself. Our goal in this method is to make sure that the credentials are working, not to worry about actually pulling data.

Let me know if that doesn't make sense for some reason and we can move forward without changing anything.

**Review by cgardens — APPROVED** [review]
Looks great. I had a couple tiny comments. I also have one question about the implementation of `check`. Let me know if my comment about check doesn't make sense, then we will merge as is!

File: airbyte-integrations/connectors/source-appstore-singer/source_appstore_singer/source.py (hunk 4 above)
Comment (Muriloo) [reply airbyte-integrations/connectors/source-appstore-singer/source_appstore_singer/source.py:56]: Understood, let me change this.

**Pull request merged**